src/classes/QueryBuilder.cls

```diff
@@ -2,41 +2,54 @@
 * This file is part of the Nebula Framework project, released under the MIT License.             *
 * See LICENSE file or go to https://github.com/jongpie/NebulaFramework for full license details. *
 *************************************************************************************************/
-public without sharing class QueryBuilder extends NebulaCore implements IQueryBuilder {
+public without sharing virtual class QueryBuilder extends NebulaCore implements IQueryBuilder {
 
-    private SObjectType sobjectType;
-    private Map<String, Schema.SObjectField> sobjectTypeFieldMap;
+    @testVisible private Boolean forUpdate;
     private Set<String> queryFields;
-    private Schema.FieldSet fieldSet;
-    private List<Schema.SObjectField> sobjectFieldList;
-    private List<String> whereClauseList;
-    private List<String> orderByList;
-    private Integer limitCount;
-    private Boolean forUpdate;
+    @testVisible private Integer limitCount;
+    private String query;
+    @testVisible private List<String> orderByList;
+    @testVisible private Boolean shouldAddCommonFields;
+    private Map<String, Schema.SObjectField> sobjectTypeFieldMap;
     private QueryFilterScope filterScope;
+    @testVisible private List<String> whereClauseList;
+    @testVisible private List<String> groupByList;
 
-    public QueryBuilder(Schema.SObjectType sobjectType, Schema.FieldSet fieldSet, List<Schema.SObjectField> sobjectFieldList) {
-        this.currentModule = NebulaCore.Module.REPOSITORY;
+    protected final Schema.FieldSet fieldSet;
+    protected final List<Schema.SObjectField> sobjectFieldList;
+    protected final SObjectType sobjectType;
 
-        this.sobjectType      = sobjectType;
+    public QueryBuilder(Schema.SObjectType sobjectType, Schema.FieldSet fieldSet, List<Schema.SObjectField> sobjectFieldList) {
         this.fieldSet         = fieldSet;
         this.sobjectFieldList = sobjectFieldList;
+        this.sobjectType      = sobjectType;
 
-        this.sobjectTypeFieldMap = this.sobjectType.getDescribe().fields.getMap();
-        this.queryFields         = new Set<String>();
-        this.whereClauseList     = new List<String>();
-        this.orderByList         = new List<String>();
-        this.forUpdate           = false;
+        this.shouldAddCommonFields = true;
```

Comment:
sorry.  some of the changes you are seeing (which I will roll back) are because:
a) we haven't implemented the Logger.cls yet
b) we haven't implemented guard clauses for custom settings.  

That's totally my bad and something I meant to mention earlier.